src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -6880,72 +6880,90 @@ export class StudyViewPageStore
 
     public getCustomChartDownloadData(chartMeta: ChartMeta): Promise<string> {
```

Comment:
I'm curious why this stuff takes place in "custom chart download data"?   and all the forking in here is kinda unpleasant.  like, this should be in configuration no?  should we just bite the bullet and do the refactor?

Reply:
Cancer Study chart is also treated as custom chart since it is not a clinical attribute. Yes, this needs a refactoring. May be we can merge this for now and do refactoring in next cycle - as it takes much time.